src/app/core/services/deletion.service.ts

```diff
@@ -0,0 +1,65 @@
+import { inject, Injectable } from '@angular/core';
+import { AuthenticationService } from './authentication.service';
+import { UsersService } from './users.service';
+import { GamesService } from './games.service';
+import { MoviesService } from './movies.service';
+import { AlbumsService } from './albums.service';
+import { Router } from '@angular/router';
+import { RatingsService } from './ratings.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DeletionService {
+  authenticationService = inject(AuthenticationService);
+  usersService = inject(UsersService);
+  gamesService = inject(GamesService);
+  moviesService = inject(MoviesService);
+  albumsService = inject(AlbumsService);
+  ratingsService = inject(RatingsService);
+  router = inject(Router);
+
+  async delete() {
+    const authUser = await this.authenticationService.auth.currentUser;
+
+    if (!authUser) {
+      throw new Error('Could not find user');
+    }
+
+    await this.deleteAllCollectionItems();
+    await this.usersService.delete(authUser.uid);
+    await authUser.delete();
+    this.router.navigate(['/']);
+  }
+
+  async deleteAllCollectionItems() {
```

Comment:
What a pain in the ass. Many DBs support a delete cascade I wonder if you find something with that keyword? It feels super wrong to orchestrate that from the client/frontend